### PR TITLE
Add SLURM-aware exp manager run timer

### PR DESCRIPTION
**Pull request by pzelasko**
# What does this PR do ?

Allow `exp_manager.max_time_per_run` to optionally count from the start of the current SLURM allocation so preprocessing and setup time cannot consume the checkpoint safety margin unnoticed.

**Collection**: Core / experiment manager

# Changelog

- Add the opt-in `max_time_per_run_from_slurm` configuration field.
- Read and validate the SLURM-provided `SLURM_JOB_START_TIME` UNIX timestamp.
- Check elapsed allocation time before training and refresh the offset when the monotonic training clock starts.
- Preserve the existing training-loop-based behavior by default.
- Add unit coverage for elapsed-time accounting, pre-fit expiry and checkpoint saving, invalid timestamps, opt-in compatibility, and public config wiring.
- Document configuration and checkpoint-buffer guidance.

# Usage

```yaml
exp_manager:
  max_time_per_run: 00:03:45:00
  max_time_per_run_from_slurm: true
```

# GitHub Actions CI

The changed core files and documentation should select the relevant GitHub Actions checks.

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Read and followed the contributor guidelines.
- [x] Added focused regression and edge-case tests.
- [x] Added user-facing configuration documentation.
- [x] Optional-install components and import guards are not affected.

**PR Type**:

- [x] New Feature
- [ ] Bugfix
- [x] Documentation

# Validation

- `pre-commit run`
- `pre-commit run --from-ref origin/main --to-ref HEAD`
- `pytest -q --cpu -m unit tests/core_ptl/test_ptl_stateless_timer.py` — 13 passed, 1 GPU-only test skipped
- `uv sync --locked --group docs`
- `uv run make -C docs html` — succeeded; 190 pre-existing optional-import and duplicate-toctree warnings
- `git diff --check origin/main...HEAD`

Not run: the full repository and GPU test suites, because the change is isolated to timer bookkeeping and is covered by focused CPU unit tests.

# Additional Information

No linked issue.

### CODE REVIEW

File: nemo/utils/exp_manager.py [hunk 1]
```diff
@@ -281,6 +281,8 @@ class ExpManagerConfig:
     ema: Optional[EMAParams] = field(default_factory=lambda: EMAParams())
     # Wall clock time limit
     max_time_per_run: Optional[str] = None
+    # Count from the SLURM allocation start instead of the training loop start.
+    max_time_per_run_from_slurm: Optional[bool] = False
```
Comment (blisc) [inline nemo/utils/exp_manager.py]: Let's default to True, even if it changes past behaviour
